apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-plugin/src/main/java/gov/cms/model/dsl/codegen/plugin/PoetUtil.java

```diff
@@ -193,6 +201,36 @@ public static MethodSpec createJoinPropertyGetter(
         .build();
   }
 
+  /**
+   * Generates a getter method that returns a mapping of fields by group, whose keys are the fields'
+   * names and whose values are the fields' values.
+   *
+   * @param groupedPropertiesName the shared group name of the fields
+   * @param propertyNames the fields for the given group
+   * @param getterResultType the return type of the grouped fields
+   * @return a {@link MethodSpec} that generates the getter method
+   */
+  public static MethodSpec createGroupedPropertiesGetter(
+      String groupedPropertiesName, List<String> propertyNames, TypeName getterResultType) {
+    ParameterizedTypeName returnType =
+        ParameterizedTypeName.get(MapClassName, StringClassName, getterResultType);
+    MethodSpec.Builder methodSpecBuilder =
+        MethodSpec.methodBuilder(fieldToMethodName("get", groupedPropertiesName))
+            .addModifiers(Modifier.PUBLIC)
+            .returns(returnType)
+            .addStatement("$T $L = new $T<>()", returnType, groupedPropertiesName, HashMap.class);
+    propertyNames.stream()
+        .forEach(
+            property ->
+                methodSpecBuilder.addStatement(
+                    "$L.put(\"$L\", $L())",
```

Comment:
I think it's better to use `$S` for a string literal rather than quoting a `$L`.

https://github.com/square/javapoet#s-for-strings